great_expectations/datasource/new_datasource.py

```diff
@@ -29,10 +29,11 @@ class BaseDatasource:
     def __init__(
         self,
         name: str,
-        execution_engine=None,
+        execution_engine: Optional[dict] = None,
```

Comment:
## non-blocking

Does the execution engine have an expected shape?
Can we define a `TypeDict` for it?